ggml/src/ggml-cuda/ggml-cuda.cu

```diff
@@ -1107,14 +1107,18 @@ static void ggml_cuda_op_mul_mat_cublas(
         const half alpha_f16 = 1.0f;
         const half beta_f16 = 0.0f;
 
+        cublasComputeType_t cu_compute_type = CUBLAS_COMPUTE_16F;
+        if(ggml_cuda_info().devices[ctx.device].cc == CC_CDNA)
+            cu_compute_type = CUBLAS_COMPUTE_32F;
```

Comment:
If the computation is done as 32 bit floats anyways you should be able to get a bit more performance by writing back the results as 32 bit directly instead of writing back as 16 bit and then converting to 32 bit.

Reply:
```suggestion
        cublasComputeType_t cu_compute_type = CUBLAS_COMPUTE_16F;
        if (ggml_cuda_info().devices[ctx.device].cc == CC_CDNA) {
            cu_compute_type = CUBLAS_COMPUTE_32F;
        }
```

Reply:
> If the computation is done as 32 bit floats anyways you should be able to get a bit more performance by writing back the results as 32 bit directly instead of writing back as 16 bit and then converting to 32 bit.

I hacked this in and found no measurable difference, this is no surprise as the time spent in the the kernels of ggml_cuda_op_mul_mat_cublas and ggml_cuda_mul_mat_batched_cublas is minuscule compared to the time spend in mmq. So its not worth the effort for now.